chore: add PULL REQUEST TEMPLATE for better documentation

billsomen opened the pull request:
# Description
Add `PULL-REQUEST-TEMPLATE.md` file defining the skeleton of the description of future PRs.

Fixes #https://github.com/botpress/v12/issues/1479

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Chore change (refactor and / or test changes)

# How Has This Been Tested?

No tests needed

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

billsomen committed:
chore: add PULL REQUEST TEMPLATE for better documentation

allardy commented:
Thoughts:
- "This change require a documentation update" feels redundant since you have a task in the checklist for the same thing. Type of change should be only the conventional commit ones (so the other 4 options)
- Header size should be the same for each sections (smaller size)
- Same casing ("Type of change", "How it was tested")
- Remove firmware, hardware, toolchain and sdk... 
- Replace with like bp_version, node_version, os, browser, binary or source

Ierezell commented on PULL-REQUEST-TEMPLATE.md:
```diff
@@ -0,0 +1,49 @@
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] Any dependent changes have been merged and published in downstream modules
```
- [ ] I've included pictures/gif/video if applicable to show the old and new behavior 

Or something in those lines?

Erik-J-D commented on PULL-REQUEST-TEMPLATE.md:
```diff
@@ -0,0 +1,49 @@
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
```
Should we keep this? I feel like we can accomplish the same thing with a pipeline action - I'd rather have a robot yell at us is we mess up than a checkbox

billsomen replied:
Yeah, good point

Erik-J-D commented on PULL-REQUEST-TEMPLATE.md:
```diff
@@ -0,0 +1,49 @@
+# PR Template
+
+This page list recommendations and general rules, so we can keep a clean and consistent PR description.
+
+*The template starts after the **horizontal line below***
+
+### References
+- https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/
+
+---
```
delete this stuff from the template?

billsomen replied:
Why not, I was'nt sure if it made sense to keep it!

billsomen committed:
Update PR template

- remove redundant "This change require a documentation update"
- make headers consistent
- fix casting for titles
- set test configuration to bp context
- add media checklist
- remove redundant checklist  : My changes generate no new warnings -> `this may be flagged by pipeline action`

billsomen committed:
Update PULL-REQUEST-TEMPLATE.md

- remove *unnecessary text* from the template -> keeping the template cleaner

billsomen committed:
fix typo and add missing word to PR description

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/botpress/botpress?pullRequest=5461) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/botpress/botpress?pullRequest=5461) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: billsomen<br/>:x: Bill - Work<br/><hr/>**Bill - Work** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/botpress/botpress?pullRequest=5461) it.</sub>

allardy commented:
Why the .gitignore? Instead, put the file in `.github/PULL_REQUEST_TEMPLATE.md` like our other templates

billsomen committed:
chore: move PR template to .github folder

allardy reviewed (approved):
LGTM, let's try this out

laurentlp committed:
Merge branch 'master' into billsomen_chore_add__pr_template
